tests/py2puml/inspection/test_inspectclass.py

```diff
@@ -148,3 +149,33 @@ def test_inspect_module_should_unwrap_decorated_constructor():
     # PointDecoratedWithoutWrapping UmlClass
     point_without_wrapping_umlitem: UmlClass = domain_items_by_fqn['tests.modules.withwrappedconstructor.PointDecoratedWithoutWrapping']
     assert len(point_without_wrapping_umlitem.attributes) == 0, 'the attributes of the original constructor could not be found, the constructor was not wrapped by the decorator'
+
+
+def test_inspect_module_should_find_methods():
+    """ Test that methods are detected including static methods
+
+    This test case assumes that the methods will be sorted by type as follow:
+        1a - instance methods (special methods aka "dunder")
+        1b - all other instance methods
+        2 - static methods """
```

Comment:
in between, there can be class methods too

Reply:
but as I said elsewhere, we can consider sorting methods later

Reply:
Class methods are a bit tricky, because they are specific to Python (and maybe a dozen other languages...) and do not have a formal representation in the UML-language as such. One option could be to make use of "[separators](https://plantuml.com/class-diagram#08ca680401eefe3a)" to group class methods, but it would require to start sorting methods 😜 

Using "[icons](https://plantuml.com/class-diagram#3644720244dd6c6a)" to represent a class method is not an option since these are reserved for other use.

Any reflection on that matter?

Reply:
the 2 links (for separators and icons) seem broken: they don't seem to refer to the thing you said.

However:
- I see what separators are and they could be used
- I believe the "icons" you mentioned are the one dedicated to visibility (private / public); thus I agree that we shouldn't use them for something else

As a first step, I suggest that we don't sort methods for now: this PR is already going to be big. What do you think?

Second question: should we keep the `self` and `cls` used in the methods' signature? I think we should, as they carry the characteristics of being either an instance or a class method. Static methods would have neither of them; thus we would not need to find visual things to distinguish them.

Reply:
ok: let's use the `{static}` modifier (as py2uml already uses it for class's attributes), let's sort later